### PR TITLE
DashboardScene: Panel edit search crash fix

### PR DESCRIPTION
**Pull request by torkelo**
After https://github.com/grafana/grafana/pull/82003 was reverted, I began looking at another fix. 

The render() functions cannot use hooks, and we need to use a hook for panel links so refactored that part out to a separate react component

### CODE REVIEW

**Review by dprokop — APPROVED** [review]
Sorry about that, I have a vivid memory of adding that hook there... but did not connect the dots.

**Review by axelavargas — APPROVED** [review]
🥇

File: public/app/features/dashboard/components/PanelEditor/getVisualizationOptions.tsx [hunk 1]
```diff
@@ -229,7 +229,8 @@ export function getVisualizationOptions2(props: OptionPaneRenderProps2): Options
         render: function renderEditor() {
           const onChange = (v: unknown) => {
             panel.onFieldConfigChange(
-              updateDefaultFieldConfigValue(currentFieldConfig, fieldOption.path, v, fieldOption.isCustom)
+              updateDefaultFieldConfigValue(currentFieldConfig, fieldOption.path, v, fieldOption.isCustom),
+              true
```
Comment (torkelo) [inline public/app/features/dashboard/components/PanelEditor/getVisualizationOptions.tsx:233]: @dprokop I had to switch replace: true here, as the "replace" false logic does handle emtpy arrays a bit strangely. 

If you pass an empty array for links for example, and let replace be false then the merge operation will add back the current links so you cannot clear/empty an array when your passing false (or leaving out replace param). So in main right now you cannot remove an item from an array (links for example, or value mappings). 
 
Feels like this merge behavior is not what we intended? 
https://github.com/grafana/scenes/blob/main/packages/scenes/src/components/VizPanel/VizPanel.tsx#L259

Comment (dprokop) [reply public/app/features/dashboard/components/PanelEditor/getVisualizationOptions.tsx:233]: Indeed, the `replace` is aimed primarily at changing the field config while defaulting things that are not passed in the partial config. Maybe there's a better way to handle that.

Comment (torkelo) [reply public/app/features/dashboard/components/PanelEditor/getVisualizationOptions.tsx:233]: @dprokop created an issue in scenes repo to investigate it at some later time :) https://github.com/grafana/scenes/issues/607

**Pull request merged**